### DISCUSSION TITLE
The History of D Machines

### DISCUSSION THREAD

**Discussion by stumbo** [Interlisp/medley#1873] [General]
The Interlisp.org webpage had a stub page that was envisioned to be the history of D-Machines.  It's had been hanging out there for several years, I've moved it to draft - effectively hiding it, with the intent to create some reasonable content and republishing the page.

As a starting place, I asked ChatGPT to write an outline with the following prompt:   _I want an outline of the history of the PARC d machine family_

It created a reasonable start.  The following is the generated response.  At present, I haven't edited it.  I like the general flow, but the first section needs to focus on the purpose, the raison d'etre, of the D machines and not become a treatise on PARC.  My working experience was with D-Lions and Daybreaks using XDE and Viewpoint, so I'm not going to provide great insight from the Interlisp perspective.  Hopefully others will.  I'm hoping you can provide deeper insight, corrections, color commentary that will help me take this start and turn it into something that speaks more to Interlisp and its relationship to the D machine family.

Here is the response:
---

The Xerox PARC “D-machine” family represents a significant chapter in computing history, with innovations at Xerox PARC (Palo Alto Research Center) that influenced later computer systems and technologies. Here’s an outline of the history of the PARC D-machine family:
1. Background and Origins (Early 1970s)

    Establishment of Xerox PARC: Founded in 1970, Xerox PARC quickly became a hub for innovation in personal computing and object-oriented programming.
    The Alto Project: Though not a D-machine, the Xerox Alto, developed in 1973, laid foundational principles for personal computers, including the GUI (graphical user interface) and Ethernet networking.

2. Conceptualizing the “D-Machine”

    Shift Toward Specialized Machines: Researchers at PARC started envisioning more specialized, high-performance workstations to advance software development, especially for the emerging field of object-oriented programming.
    Introduction of Interlisp and Smalltalk: These programming languages emphasized structured, interactive, and object-oriented programming paradigms that required optimized hardware.

3. The Dorado (Late 1970s - Early 1980s)

    Dorado Development: The first D-machine, the Dorado, was a high-performance workstation designed to handle intensive Smalltalk and Interlisp-D applications.
    Technical Specifications: Featuring a bit-slice architecture and a speed significantly faster than the Alto, the Dorado offered improved graphics and computational power.
    Impact on Development: It served as a powerful platform for developing sophisticated applications and system software, influencing later workstation designs.

4. The Dolphin and Dandelion (Early 1980s)

    Dolphin: This was a less powerful and more affordable workstation compared to the Dorado, designed to handle day-to-day programming tasks in Interlisp and Smalltalk environments.
    Dandelion: Released as part of the Xerox 1100 series, the Dandelion was used to run the Viewpoint (later GlobalView) office environment, aimed at broader office productivity use beyond programming.
    Dandelion’s Role: The Dandelion also helped bridge the gap between PARC’s advanced computing concepts and Xerox’s business products.

5. Later D-Machines and Their Legacy

    Daybreak: The Xerox Daybreak (Xerox 6085) was an extension of the D-machine family, intended to be a more user-friendly workstation with improved graphics capabilities.
    Influence on Future Workstations: The D-machine family influenced the development of workstation computing, particularly the integration of GUIs and networking capabilities.
    Impact on Commercial Products: Elements pioneered by the D-machine family found their way into commercial products and standards, especially through the ideas carried by former PARC employees who moved to Apple, Microsoft, and other tech companies.

6. Legacy and Influence on Modern Computing

    Networking and GUI: The D-machines’ incorporation of Ethernet and GUI principles influenced the design of personal computers and workstations.
    Object-Oriented Programming: The D-machine family’s support for languages like Smalltalk accelerated the adoption of object-oriented programming principles that now underlie much of modern software development.
    Technology Transfer: Although Xerox did not commercialize many of these technologies successfully, the D-machine family’s concepts were critical to the development of products like Apple’s Lisa and Macintosh computers.

The PARC D-machines laid a foundation for innovations in personal computing, networking, and graphical interfaces, with their influence visible in many of today’s computing technologies.

**Comment by timdthomas** [comment]
Hey Bill!  I don't recall you being involved in interlisp at Xerox!  Not sure how I stumbled across this, but in my retirement I was recalling the Interlisp/Trillium days and how much ahead of the industry it was, and thought I wonder if it's still around, and here I am.  I've forgotten pretty much everything Interlisp related, wonder if it will come back if I install and play?

**Reply by nbriggs** [reply]
@timdthomas - you can install and play, or just visit https://online.interlisp.org/

I haven't resurrected Trillium, but at least the Birthday84 sources are in the PARC archives online at the Computer History Museum...

**Reply by timdthomas** [reply]
I tried online, and my mind completely went blank!  I lived and breathed Interlisp for 5 years, but it's been over 30 years since I've used it.  I couldn't even figure out how to start SEdit.  For years after leaving the interlisp world, I missed SEdit as there was nothing like it for other languages.   I did enough to send a screenshot to my wife (also a former Xerox interlisp person, though mostly a UI designer using Trillium.)

**Reply by nbriggs** [reply]
In an Interlisp exec, `(DF FOO)` and it'll pop up a menu asking whether you want a FNS or FUNCTIONS dummy definition for FOO.  Have fun.  I'm trying to place you, but my memory for all the names that long ago is not the best!  Would I remember your wife's name?

**Reply by timdthomas** [reply]
I was on the Docutech Trillium team (UIDS), my wife (Carol Parsons at the time) was a UI designer.  We had 2 different Trillium teams, 5090 used a different version of Trillium.  Same concepts, but very different way of using it.

**Reply by nbriggs** [reply]
Ah, the names that come to mind are Pat Hannaway and Kathy Matysek (weren't they both on the DT side?) I also remember Gary Gocek and Tim Bigham.

I wonder if any of the old Trillium interfaces out of IDHI/UIDS got saved in a way we could get them going again!

**Reply by pamoroso** [reply]
@timdthomas What worked for me as an Interlisp newcomer is the [Medley primer](https://interlisp.org/documentation/Medley-Primer.pdf).

**Reply by pamoroso** [reply]
@timdthomas For the Medley project we hold weekly online meetings on Wednesdays at 10:00 am PT. It would be great if you could join one such meeting to share your experience and have a chat with other Xerox alumni (not me).
